src/svg2pdf.js

```diff
@@ -151,16 +153,103 @@ SOFTWARE.
     return pathSegList;
   };
 
+  function StyleSheets(rootSvg, loadExtSheets) {
+    this.rootSvg = rootSvg;
+    this.loadExtSheets = loadExtSheets;
+    this.sheets = [];
+    this.selectorMap = {};
+  }
+
+  StyleSheets.prototype.getParsedSheets = function () {
+    if (this.sheets.length === 0) {
+      var sheetTexts = [];
+
+      var styleTags = this.rootSvg.getElementsByTagName("style") || [];
+      for (styleTag of styleTags) {
+        sheetTexts.push(styleTag.textContent);
+      }
+
+      if (this.loadExtSheets || true) {
+        var paths = [];
+        var links = this.rootSvg.getElementsByTagName("link");
+        for (var link of links) {
+          if (link.getAttribute("rel") === "stylesheet") {
+            paths.push(link.getAttribute("href"));            
+          }
+        }
+
+        for (var node of this.rootSvg.ownerDocument.childNodes) {
+          if (node.nodeName === "xml-stylesheet") {
+            paths.push(node.data.match(/href=[\"|\'|\`|\`].*?[\"|\'|\`|\`]/)[0].split("=")[1].slice(1,-1));
+          }
+        }
+
+        for (var path of paths) {
+          var response = StyleSheets.loadSheet(path);
+          if (!(response instanceof Error)) {
+            sheetTexts.push(response);
+          }
+        }
+      }
+
+      for (var text of sheetTexts) {
+        var sheet = Css.parse(text, {silent: true});
+        this.sheets.push(sheet);
+      }
+    }
+    return this.sheets;
+  }
+
+  StyleSheets.loadSheet = function (url) {
+    const request = new XMLHttpRequest();
+    request.open('GET', url, false);
+    request.overrideMimeType('text\/plain; charset=x-user-defined');
+    request.send();
+
+    if (request.status !== 200) {
+      return new Error(`Unable to fetch ${url}, status code: ${request.status}`);
+    }
+
+    return request.responseText;
+  }
+
+  StyleSheets.prototype.getRuleFor = function (node, propertyCss) {
+    for (var sheet of this.getParsedSheets()) {
+      for (var rule of sheet.stylesheet.rules) {
+        for (var selector of rule.selectors) {
+          if (this.getCompiledRule(selector)(node, selector)) {
+            for (var declaration of rule.declarations) {
+              if (declaration.property === propertyCss) {
+                return declaration.value;
+              }
+            }            
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  StyleSheets.prototype.getCompiledRule = function(selector) {
+    return this.selectorMap[selector] || (this.selectorMap[selector] = CSSselect.compile(selector, {adapter: BrowserDomAdapter}));      
+  }
+
   // returns an attribute of a node, either from the node directly or from css
-  var getAttribute = function (node, propertyNode, propertyCss) {
-    propertyCss = propertyCss || propertyNode;
+  var getAttribute = function (node, propertyNames, styleSheets) {
```

Comment:
we could define getAttribute on the context object, as we need a context anyways for this.

Reply:
Actually I tried to implement it in a way that you can use getAttribute() without any context...